### PR TITLE
Use BQ CSV loader

### PR DESCRIPTION
**Pull request by ghost**
Allows you to load raw `jsonb` columns out of postgres as strings, without knowing their schema.

Eyeball tests look like the CSV loader is faster than json, this should be the case both for reading out of pgsql and for pushing to BQ as there is less data to wrangle.

Also handle UUID postgres types as strings.

Special use case I have is the use of 2 columns `md_insert` and `md_update` that are stored as int and hold epoch value.  If you push epoch to a timestamp column in BQ it converts these to proper timestamp automatically which is quite useful.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
This is going to conflict with https://github.com/mattrobenolt/pgtobq/pull/4 at the moment, and I haven't taken the time to wrap my head around this yet. Why do we need to switch to using CSV instead of JSON?

**Comment by ghost** [comment]
I just saw that PR and realised the same thing!  The main reason I have for the CSV loader was as I am storing `jsonb` in pgsql and want to store it in bq.  In theory you should be able to load nested json into bq but it looks like you need to define the schema for the nested objects and you can't derive these from the information_schema table as the db doesn't store anything about jsonb columns.

**Comment by mattrobenolt** [comment]
🤔 I'm not sure that's true, I think there's just a "Record" type, and I don't think you need to store the whole schema for that. I think that's the point, but I could be wrong, we don't have this use case ourselves, so I'm open to hearing ideas.

**Comment by mattrobenolt** [comment]
Oh, I think you're right. A Record needs a bunch of sub fields.

So are you suggesting that you'd store a jsonb column as just TEXT?

**Comment by ghost** [comment]
I am happy to be wrong about it :-), I played with record type for a bit but it wanted you to define the type of each sub element.  You can see this in the UI when you try and create a table manually.  Docs here https://cloud.google.com/bigquery/docs/nested-repeated seem to indicate the need of a record type schema.

**Comment by ghost** [comment]
Basically, yes. It looks a little ugly to store the object as a string but BQ comes with some helper functions to extract the data, similar to pgsqls json functions.  https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#json_extract-or-json_extract_scalar 
`JSON_EXTRACT_SCALAR('{ "name" : "Jakob", "age" : "6" }', '$.name') as scalar_name,` 

My thinking was, because I am using the jsonb as dumping grounds in my tables, I don't mind too much if I need to write a view ontop of the bq table to extract the required fields.  Lets me be flexible on input and update my view as needed.

**Comment by mattrobenolt** [comment]
Yeah, I agree with that, I don't thinkw e should try to store jsonb as structured RECORD type in BQ. So a string in JSON format seems fine to me.

I assume the problem is that dumping to JSON with `row_to_json` just gives us a nested object instead of a string for the field. I wonder if this behavior can be changed...

**Comment by ghost** [comment]
I can merge in your udt_name change to this, will work with this.

Yes, the issue is with `row_to_json` I looked at a couple options but couldn't see a way around it, short of writing a custom query to extract the jsonb fields which rather defeats the purpose of the loader.

**Comment by mattrobenolt** [comment]
Ok, let's break this up into two discrete changes here so it's easier for me:

* Adding your new types, based on my other open PR (I'll merge this now)
* Changing to CSV for loading. I'll need to manually test this and whatnot to make sure this is compatible with what we're doing.

**Comment by ghost** [comment]
I updated my code to use udt_name, all works for me.  Take your time retesting.  In general though I think you should only see performance improvements, whether you are taking advantage of jsonb lodaing or not.  Along the way I had few bugs when switching but I believe I caught them all.

File: main.go [hunk 1]
```diff
@@ -37,20 +39,28 @@ func (c *Column) ToFieldSchema() *bigquery.FieldSchema {
 	f.Name = c.Name
 	f.Required = c.IsNullable == "NO"
 
+	switch c.Name {
+	// Allow BQ to convert special Epoch columns to timestamp
```
Comment (mattrobenolt) [inline main.go]: @sbramin wait, is this a special column name in your project? Or why is this being done this way based on the column name itself?

What is the actual column type according to `information_schema`?

Comment (ghost) [reply main.go]: Had forgotten about that.  The columns are stored as integers in pgsql and hold epoch values.  Both pgsql and BQ have a little magic which lets you insert epoch values into timestamp columns and it automatically renders them as correct timestamp values. For my apps I prefer reading/writing in epoch but once it gets to BQ other people are less keen so I take advantage of this auto converting.  

I'll remove this / perhaps make it a flag.

**Comment by ghost** [comment]
Hey, did you ever get round to looking at loading the jsonb fields?  I've fixed the conflicts after your merge.

**Pull request closed without merging**